clang/lib/StaticAnalyzer/Checkers/WebKit/PtrTypesSemantics.cpp

```diff
@@ -356,6 +364,14 @@ class TrivialFunctionAnalysisVisitor
     return TrivialFunctionAnalysis::isTrivialImpl(Callee, Cache);
   }
 
+  bool VisitCXXDefaultArgExpr(const CXXDefaultArgExpr *E) {
```

Comment:
In theory this may need to be cached because it may cause default argument expressions to be visited multiple times, i.e. every time a function is called with the given argument omitted. And these expressions can get quite complicated.

I haven't ever seen it matter though.

Reply:
Yeah, I'm adding more general caching mechanism for `Stmt` in https://github.com/llvm/llvm-project/pull/82229.